src/main/java/pl/edu/pw/pik/pikactivitytrackerserver/Repository/UserRepository.java

```diff
@@ -1,8 +1,18 @@
 package pl.edu.pw.pik.pikactivitytrackerserver.Repository;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
 import pl.edu.pw.pik.pikactivitytrackerserver.model.User;
 
+import java.util.List;
+
+@Repository
 public interface UserRepository extends JpaRepository<User, Integer> {
 
+//    User findByUsername(String username);
+
+    @Query(value = "select * from users where username = ?", nativeQuery = true)
+    List<User> findByUsername(String username);
```

Comment:
Rozważyłbym zmianę nazwy funkcji, żeby podkreślić, że zwraca AllUserName. Druga sprawa, czy ona ma sens w tej postaci? I mean, jeśli dobrze pamiętam to w bazie mamy ograniczenie unique na username, więc nie widzę potrzeby zwracania list - można tylko user

Reply:
tak, to prawda że jest tylko jeden użytkownik o danej nazwie, zmieniłam na sam user, przetestowane, zmian dokonałam już w logowaniu